### PR TITLE
🐛 test/e2e/in-memory: set providerID after VM is provisioned

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
This PR changes the in-memory provider to set the providerID already after the VM is provisioned not after the Node is ready.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go [hunk 1]
```diff
@@ -263,12 +264,14 @@ func (r *InMemoryMachineReconciler) reconcileNormalCloudMachine(ctx context.Cont
 		x := inMemoryMachine.Spec.Behaviour.VM.Provisioning
 
 		provisioningDuration = x.StartupDuration.Duration
-		jitter, err := strconv.ParseFloat(x.StartupJitter, 64)
-		if err != nil {
-			return ctrl.Result{}, errors.Wrapf(err, "failed to parse VM's StartupJitter")
-		}
-		if jitter > 0.0 {
-			provisioningDuration += time.Duration(rand.Float64() * jitter * float64(provisioningDuration)) //nolint:gosec // Intentionally using a weak random number generator here.
+		if x.StartupJitter != "" {
```
Comment (sbueringer) [inline test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go:267]: Unit tests were failing locally without handling empty string here explicitly

Comment (sbueringer) [reply test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go:267]: I'll enable the unit tests in CI and fix some more issues..

**Review by fabriziopandini — COMMENTED** [review]
thanks for the fix!
/lgtm
/approve

**Pull request merged**